lib/data_validator.rb

```diff
@@ -49,32 +49,27 @@ class DataValidator
                      '2.16.840.1.114222.4.11.1066',
                      '2.16.840.1.113883.1.11.19579'].freeze
 
-    def initialize(bundle, measure_ids)
-      @bundle = bundle
-      measures = @bundle.measures.in(hqmf_id: measure_ids)
-
-      @oids = measures.collect(&:oids).flatten.uniq + HL7_QRDA_OIDS
+    def initialize(measure_ids)
+      measures = CQM::Measure.find(measure_ids)
+      @value_set_ids = measures.collect { |m| m.value_sets.map(&:_id) }.flatten.uniq
     end
 
     def validate(file, options = {})
       doc = get_document(file)
 
       doc.xpath('//*[@sdtc:valueSet]').each_with_object([]) do |node, errors|
         oid = node.at_xpath('@sdtc:valueSet')
-        vs = @bundle.value_sets.where('oid' => oid).first
+        vs = CQM::ValueSet.where('oid' => oid, '_id' => { '$in' => @value_set_ids }).first
+        next unless vs
```

Comment:
It looks like not finding the valueset used to be an error. Should this still be an error?

Reply:
This error was then later ignored by Cypress.  Gone are the days of a restrictive 'Scoop and Filter' where one was required to filter out information not related to a measure.